albinos-cli/source/cli/cli.d

```diff
@@ -0,0 +1,123 @@
+module cli.cli;
+
+import core.exception : AssertError;
+import std.stdio;
+import std.string;
+import std.socket : SocketOSException;
+import std.getopt;
+import std.uni : isWhite;
+import std.process : executeShell;
+import std.file;
+import asdf;
+import protocol.protocol_type;
+import client.client;
+import cli.deserializer;
+
+class CLI
+{
+    this(string socket_path)
+    {
+        try
+        {
+            client_ = new Client(socket_path);
+        }
+        catch (SocketOSException e)
+        {
+            writeln(stderr, e.msg);
+            is_running_ = false;
+        }
+    }
+
+    private void load_config(string[] args) 
+    in
+    {
+        assert(args !is null, "args cannot be null");
+        assert(args.length == 2, "need 1 arguments");
```

Comment:
```suggestion
        assert(args.length == 2, "needs 1 arguments");
```
(also doesn't this technically need 2 arguments?)

Reply:
Splitted_line contains the command name and the args, (requirement for getopt)

Reply:
IIRC getops also accepts `--arg value` on top of `--arg=value`. I'm fine with the wording anyway.